### ISSUE TITLE
Port the preptools to a modern oxygen

### ISSUE THREAD

**Issue by egli** [sbsdev/dtbook-preptools#2]
The current preptools are built against a very old version of the oxygen editor. 
- [x] Get the current SDK
- [x] Port the ant build script to maven
- [x] Port the current implementation to the new SDK

**Comment by egli** [comment]
See https://www.oxygenxml.com/doc/versions/21.0/ug-editor/topics/extend-oxygen-with-plugins.html

**Comment by egli** [comment]
The current preptools fail when deployed in oxygen 21
- they show up in the UI and in the menu
- but are disabled
- there is also a NPE inside DocumentMetaInfo.setCurrentState
```
34187 FATAL [ AWT-EventQueue-0 ] ro.sync.exml.workspace.b.c.d - java.lang.NullPointerException
java.lang.NullPointerException
	at ch.sbs.plugin.preptools.DocumentMetaInfo.setCurrentState(DocumentMetaInfo.java:293)
	at ch.sbs.plugin.preptools.PrepToolsPluginExtension$2.editorPageChanged(PrepToolsPluginExtension.java:462)
	at ro.sync.exml.workspace.b.c.d$_b.editorPageChanged(Unknown Source)
	at ro.sync.exml.editor.gc.m(Unknown Source)
	at ro.sync.exml.editor.zd$34.run(Unknown Source)
	at ro.sync.ui.me$1.run(Unknown Source)
	at ro.sync.ui.me.i(Unknown Source)
	at ro.sync.exml.editor.zd.editorPageChanged(Unknown Source)
	at ro.sync.exml.editor.vc$11.run(Unknown Source)
	at ro.sync.ui.me$1.run(Unknown Source)
	at ro.sync.ui.me.i(Unknown Source)
	at ro.sync.exml.editor.vc.changePage(Unknown Source)
	at ro.sync.exml.editor.vc.changePage(Unknown Source)
	at ro.sync.exml.editor.validate.r.changePage(Unknown Source)
	at ro.sync.exml.editor.vc.changePageOnButtonSwitch(Unknown Source)
	at ro.sync.exml.editor.ee$1.itemStateChanged(Unknown Source)
```

**Comment by egli** [comment]
It might make sense to port to Maven first, as with Maven you can easily and still pick old versions of the SDK to build against

**Comment by egli** [comment]
Fixed in bdb9307fd70413330c9b480b536e73a7bffb3b66
